### ISSUE TITLE
Make sure SpectralCoord implementation properly supports relativisic velocities

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#10249]
Currently the implementation of SpectralCoord from https://github.com/astropy/astropy/pull/10185 does correctly support inputting relativistic redshifts and converting them to the 'correct' velocities, and we do take care in a few places to add velocities using the relativistic velocity addition formula. However, there are still some cases (e.g. https://github.com/astropy/astropy/pull/10185#discussion_r418511614) where we should be using the relativistic approach too - the remaining cases appear to be mostly related to when the observer and target are relativistic.

The bottom line is that we should go through the code in SpectralCoord carefully to determine which operations need to be updated, and add appropriate tests. Because the differentials framework in astropy.coordinates does not support relativistic calculations, we will have to hard-code some operations that we currently rely on differentials for.

Ideally it would be good to try and fix this before the final 4.1 release, but in case we don't manage, I have added a warning in the docs in #10185 regarding some unsupported relativistic cases.

**Comment by mhvk** [comment]
Probably best done as a little helper function that can deal with representations as well. In fact, would be quite useful utility anyway...

**Comment by mhvk** [comment]
It's clear we didn't get to this, so clearing milestone. For now, the warning will have to do...
